### PR TITLE
Update to coding-standard 1.2.3

### PR DESCRIPTION
**Pull request by kesselb**
* Resolves: unblock dependabot update for coding-standard 1.2.3

## TODO

- [x] Update OpenApi definitions
- [x] Make Psalm happy
- [x] Add SPDX for .git-blame-ignore-revs
- [x] Update check for expected files
- [x] Ping release team to ignore .git-blame-ignore-revs

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by nickvergessen — APPROVED** [review]

**Review by provokateurin — APPROVED** [review]

**Comment by kesselb** [comment]
> Error: apps/files_versions/appinfo/routes.php:22:1: InvalidScope: Use of $this in non-class context (see https://psalm.dev/013)

False positive (because the file is included)

> Error: lib/private/Route/Router.php:177:7: NoInterfaceProperties: Interfaces cannot have properties (see https://psalm.dev/028)
Error: lib/private/Route/Router.php:180:4: NoInterfaceProperties: Interfaces cannot have properties (see https://psalm.dev/028)
Error: lib/private/Route/Router.php:182:3: NoInterfaceProperties: Interfaces cannot have properties (see https://psalm.dev/028)

I agree with Psalm the code is highly confusing.

**Comment by susnux** [comment]
>  ERROR: There were 1 additional files:
> .git-blame-ignore-revs
> ERROR: Please remove or add those files again or inform the release team about those now files to be included or excluded from the release tar ball.

Should be excluded from release tar ball, no?

**Comment by nickvergessen** [comment]
yes should be removed on packaging

File: .git-blame-ignore-revs [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# .git-blame-ignore-revs
```
Comment (st3iny) [inline .git-blame-ignore-revs:1]: TIL!

**Review by st3iny — APPROVED** [review]
Looks good!

**Comment by kesselb** [comment]
> Error: apps/files_versions/appinfo/routes.php:22:1: InvalidScope: Use of $this in non-class context (see https://psalm.dev/013)
> Error: lib/private/Route/Router.php:177:7: NoInterfaceProperties: Interfaces cannot have properties (see https://psalm.dev/028)
> Error: lib/private/Route/Router.php:180:4: NoInterfaceProperties: Interfaces cannot have properties (see https://psalm.dev/028)
> Error: lib/private/Route/Router.php:182:3: NoInterfaceProperties: Interfaces cannot have properties (see https://psalm.dev/028)

The above errors were fixed by changing `@var $this \OCP\Route\IRouter` to  `@var $this \OC\Route\%outer` in `core/routes.php`. 

https://github.com/nextcloud/server/blob/1cc6b3577fdbeadece7e4e6478e7f7755555b41a/lib/private/Route/Router.php#L166-L182

Psalm resolves the require_once call and assumes, because of the phpdoc annotation, the the class is now IRouter instead of Router and shows a warning that the interface does not have loaded, etc..

**Comment by kesselb** [comment]
Cypress is having a good day :partying_face: :raised_hands:

**Pull request merged**